### PR TITLE
HQ Hunter fix

### LINKED ISSUE #4681: HQ_Hunter does not destroy conquered military sites of other tribe
Sorry, now i found a bug in hq_hunter:

If a player had conquered a military building wich belongs to a different tribe than the player plays, it get not destroyed if his headquarter was destroyed.

Difficult to reproduce... i have a replay but it does not work because [i used an addon which does not work for replays](https://github.com/widelands/wl_addons_server/issues/21)

Instead of a replay i try to explain:
Player A plays barbarians
Player B plays empire

1. Player A conquers a military site of player B, say 'empire_sentry'
2. Player B destroys A's headquarters

The conquered military site 'empire_sentry' from 1 will then not destroyed because it is not part of barbarians buildings.

I guess the culprit is in [check_player_defeated](https://github.com/widelands/widelands/blob/master/data/scripting/win_conditions/win_condition_functions.lua#L75)

where only buildings gathered a players tribe has.

### PR DESCRIPTION
**Pull request by hessenfarmer**
fixes #4681
adds documentation to start conditions

### CODE REVIEW

File: data/tribes/initialization/atlanteans/starting_conditions/discovery.lua [hunk 1]
```diff
@@ -1,3 +1,102 @@
+-- RST
+-- .. _lua_tribes_tribes_start_conditions:
+--
+-- Start Conditions
+-- ================
+--
+-- The start conditions a tribe can use are listed in the tribes :ref:`init.lua <lua_tribes_tribes_init>` file.
+--
+-- They are defined in their defintion files named as follows and located in
+-- ``data/tribes/initialization/<tribe_name>/starting_conditions/<start_condition_name.lua``.
```
Comment (Noordfrees) [inline data/tribes/initialization/atlanteans/starting_conditions/discovery.lua]:
```suggestion
-- They are defined in their definition files named as follows and located in
-- ``data/tribes/initialization/<tribe_name>/starting_conditions/<start_condition_name>.lua``.
```

File: data/tribes/initialization/atlanteans/starting_conditions/discovery.lua [hunk 2]
```diff
@@ -1,3 +1,102 @@
...
+--
+-- .. function:: init
+--
+--    This function initializes the start condition.
+--
+--    :arg table: This table contains all the data needed to initilize the start condition.
```
Comment (Noordfrees) [inline data/tribes/initialization/atlanteans/starting_conditions/discovery.lua]:
```suggestion
--    :arg table: This table contains all the data needed to initialize the start condition.
```

File: data/tribes/initialization/atlanteans/starting_conditions/discovery.lua [hunk 3]
```diff
@@ -1,3 +1,102 @@
...
+--                It contains the following entries:
+--
+--    **descname**: The localized name of the start condition.
+--
+--    **tooltip**: A short description of the start condition. Should contain essential hints if necessary.
+--
+--    **map_tags**: *Optional* A table with map tags that are needed for this startcondition.
+--                  e.g. ``{ "seafaring" }`` if the start conditions needs a seafaring map
```
Comment (Noordfrees) [inline data/tribes/initialization/atlanteans/starting_conditions/discovery.lua]:
```suggestion
--    **map_tags**: *Optional* A table with map tags that are required for this startcondition.
--                  e.g. ``{ "seafaring" }`` if the start conditions needs a seafaring map.
--                  If present, the start condition can only be selected by the user if the map contains *all* listed tags.
```

File: data/tribes/initialization/atlanteans/starting_conditions/discovery.lua [hunk 4]
```diff
@@ -1,3 +1,102 @@
...
+--
+--    **incompatible_wc**: *Optional* A table with strings containing the path of win conditions taht are incompatible with this
+--                         startcondition.
+--                         e.g. ``{ "scripting/win_conditions/hq_hunter.lua" }`` if the start conditions is incompatible
+--                         with the win condition "HQ Hunter"
```
Comment (Noordfrees) [inline data/tribes/initialization/atlanteans/starting_conditions/discovery.lua]:
```suggestion
--    **incompatible_wc**: *Optional* A table with strings containing the paths of win conditions that are incompatible with this
--                         start condition, e.g. ``{ "scripting/win_conditions/hq_hunter.lua" }`` if the start condition
--                         is incompatible with the win condition "HQ Hunter".
```

File: data/tribes/initialization/atlanteans/starting_conditions/discovery.lua [hunk 5]
```diff
@@ -1,3 +1,102 @@
...
+--
+--    **func**: A standardized function allowing to share this start conditions between players (Shared Kingdom)
```
Comment (Noordfrees) [inline data/tribes/initialization/atlanteans/starting_conditions/discovery.lua]: Needs more information. What does this standard require about this function?

File: data/tribes/initialization/atlanteans/starting_conditions/discovery.lua [hunk 6]
```diff
@@ -1,3 +1,102 @@
...
+--
+-- After these declarations the lua commands to define the initial mapobjects for the player follow.
+--
+-- For making the UI texts translateable, we also need to push/pop the correct textdomain.
```
Comment (Noordfrees) [inline data/tribes/initialization/atlanteans/starting_conditions/discovery.lua]:
```suggestion
-- To make the UI texts translatable, we also need to push/pop the correct textdomain.
```

**Comment by Noordfrees** [comment]
We use the terms "Start condition" and "Starting condition" interchangeably currently. This should be unified. Which term do we want? I think "Starting condition" is used in more places than "Start condition".

**Comment by Noordfrees** [comment]
I cannot find a link to the new documentation from the HTML file tree. I think this needs a cross-reference from autogen_toc_lua_tribes_defining or such and an entry in some TOC.
```
checking consistency... /home/benedikt/wl/widelands/doc/sphinx/source/autogen_auxiliary_mapobjects.rst: WARNING: document isn't included in any toctree
/home/benedikt/wl/widelands/doc/sphinx/source/autogen_lua_tribes_defining_atlanteans.rst: WARNING: document isn't included in any toctree
/home/benedikt/wl/widelands/doc/sphinx/source/autogen_lua_tribes_defining_preload.rst: WARNING: document isn't included in any toctree
/home/benedikt/wl/widelands/doc/sphinx/source/autogen_lua_tribes_other_global_helptexts.rst: WARNING: document isn't included in any toctree
```

**Comment by hessenfarmer** [comment]
> 
> 
> I cannot find a link to the new documentation from the HTML file tree. I think this needs a cross-reference from autogen_toc_lua_tribes_defining or such and an entry in some TOC.
> 
> ```
> checking consistency... /home/benedikt/wl/widelands/doc/sphinx/source/autogen_auxiliary_mapobjects.rst: WARNING: document isn't included in any toctree
> /home/benedikt/wl/widelands/doc/sphinx/source/autogen_lua_tribes_defining_atlanteans.rst: WARNING: document isn't included in any toctree
> /home/benedikt/wl/widelands/doc/sphinx/source/autogen_lua_tribes_defining_preload.rst: WARNING: document isn't included in any toctree
> /home/benedikt/wl/widelands/doc/sphinx/source/autogen_lua_tribes_other_global_helptexts.rst: WARNING: document isn't included in any toctree
> ```

Well I hoped to achieve this with the entry in extract_rst.py as the TOC is autogenerated from this as far as I understand it.
Will try to setup the doc building in my MSYS environment to investigate

**Comment by frankystone** [comment]
Please add also a link in [Add-ons starting condition](https://www.widelands.org/documentation/add-ons/#starting-condition) which links to your new documentation.

**Comment by frankystone** [comment]
If you want to address unifying of 'start condition' vs. 'starting condition' in this branch here is a list.

The number after the filename is the line number:
~~~~
git branch: (3849_win_conditions)$:> grep -rni 'start condition' data/
data/tribes/initialization/frisians/starting_conditions/headquarters.lua:2:--                 Start conditions for Headquarters
data/tribes/initialization/frisians/starting_conditions/discovery.lua:2:--                 Start conditions for Discovery
data/tribes/initialization/frisians/starting_conditions/new_world.lua:2:--                 Start conditions for New World
data/tribes/initialization/amazons/starting_conditions/headquarters.lua:2:--                 Start conditions for Headquarters
data/tribes/initialization/empire/starting_conditions/discovery.lua:2:--                 Start conditions for Discovery
data/tribes/initialization/empire/starting_conditions/new_world.lua:2:--                 Start conditions for New World
data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:4:-- Start Conditions
data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:7:-- The start conditions a tribe can use are listed in the tribes :ref:`init.lua <lua_tribes_tribes_init>` file.
data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:14:--    This function initializes the start condition.
data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:16:--    :arg table: This table contains all the data needed to initialize the start condition.
data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:19:--    **descname**: The localized name of the start condition.
data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:21:--    **tooltip**: A short description of the start condition. Should contain essential hints if necessary.
data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:24:--                  e.g. ``{ "seafaring" }`` if the start conditions needs a seafaring map.
data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:25:--                  If present, the start condition can only be selected by the user if the map contains *all* listed tags.
data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:28:--                         start condition, e.g. ``{ "scripting/win_conditions/hq_hunter.lua" }`` if the start condition
data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:31:--    **func**: A standardized function to determine whether to share this start condition between players (Shared Kingdom)
data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:34:--              together with the start condition.
data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:111:--                 Start conditions for Discovery
data/tribes/initialization/atlanteans/starting_conditions/new_world.lua:2:--                 Start conditions for New World
data/tribes/initialization/barbarians/starting_conditions/headquarters.lua:2:--                 Start conditions for Headquarters
data/tribes/initialization/barbarians/starting_conditions/discovery.lua:2:--                 Start conditions for Discovery
data/tribes/initialization/barbarians/starting_conditions/fortified_village.lua:2:--                    Start conditions for Fortified Village
data/tribes/initialization/barbarians/starting_conditions/village.lua:2:--                    Start conditions for Village
data/tribes/initialization/barbarians/starting_conditions/new_world.lua:2:--                 Start conditions for New World
~~~~

**Comment by hessenfarmer** [comment]
> 
> 
> If you want to address unifying of 'start condition' vs. 'starting condition' in this branch here is a list.
> 
> The number after the filename is the line number:
> 
> ```
> git branch: (3849_win_conditions)$:> grep -rni 'start condition' data/
> data/tribes/initialization/frisians/starting_conditions/headquarters.lua:2:--                 Start conditions for Headquarters
> data/tribes/initialization/frisians/starting_conditions/discovery.lua:2:--                 Start conditions for Discovery
> data/tribes/initialization/frisians/starting_conditions/new_world.lua:2:--                 Start conditions for New World
> data/tribes/initialization/amazons/starting_conditions/headquarters.lua:2:--                 Start conditions for Headquarters
> data/tribes/initialization/empire/starting_conditions/discovery.lua:2:--                 Start conditions for Discovery
> data/tribes/initialization/empire/starting_conditions/new_world.lua:2:--                 Start conditions for New World
> data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:4:-- Start Conditions
> data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:7:-- The start conditions a tribe can use are listed in the tribes :ref:`init.lua <lua_tribes_tribes_init>` file.
> data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:14:--    This function initializes the start condition.
> data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:16:--    :arg table: This table contains all the data needed to initialize the start condition.
> data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:19:--    **descname**: The localized name of the start condition.
> data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:21:--    **tooltip**: A short description of the start condition. Should contain essential hints if necessary.
> data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:24:--                  e.g. ``{ "seafaring" }`` if the start conditions needs a seafaring map.
> data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:25:--                  If present, the start condition can only be selected by the user if the map contains *all* listed tags.
> data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:28:--                         start condition, e.g. ``{ "scripting/win_conditions/hq_hunter.lua" }`` if the start condition
> data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:31:--    **func**: A standardized function to determine whether to share this start condition between players (Shared Kingdom)
> data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:34:--              together with the start condition.
> data/tribes/initialization/atlanteans/starting_conditions/discovery.lua:111:--                 Start conditions for Discovery
> data/tribes/initialization/atlanteans/starting_conditions/new_world.lua:2:--                 Start conditions for New World
> data/tribes/initialization/barbarians/starting_conditions/headquarters.lua:2:--                 Start conditions for Headquarters
> data/tribes/initialization/barbarians/starting_conditions/discovery.lua:2:--                 Start conditions for Discovery
> data/tribes/initialization/barbarians/starting_conditions/fortified_village.lua:2:--                    Start conditions for Fortified Village
> data/tribes/initialization/barbarians/starting_conditions/village.lua:2:--                    Start conditions for Village
> data/tribes/initialization/barbarians/starting_conditions/new_world.lua:2:--                 Start conditions for New World
> ```

Many thanks. That helps a lot. and yes I definitly want to include the unified term in this branch. 
Another question is there any way to test the documentation changes in a website test?

**Comment by frankystone** [comment]
> Another question is there any way to test the documentation changes in a website test?

I did that already :-)

**Comment by frankystone** [comment]
Here are the zipped .html files including the .css made with your branch and the website code.
[current.zip](https://github.com/widelands/widelands/files/5988335/current.zip)

Just unpack it somewhere and open the file `path\current\index.html`

File: doc/sphinx/source/add-ons.rst [hunk 7]
```diff
@@ -104,7 +104,7 @@ starting_condition
 ~~~~~~~~~~~~~~~~~~
 A starting condition script. May define the same starting conditions for any number of tribes.
 
-The add-on needs to contain one or more scripts called ``<tribename>.lua`` which must follow the same conventions as the files in ``data/tribes/initialization/*/starting_conditions/*.lua``.
+The add-on needs to contain one or more scripts called ``<tribename>.lua`` which must follow the same conventions as the files in :ref:`data/tribes/initialization/*tribename*/starting_conditions/*.lua <lua_tribes_tribes_start_conditions>`.
```
Comment (Noordfrees) [inline doc/sphinx/source/add-ons.rst]: This looks very inconsistent and needs changing, e.g.:
```suggestion
The add-on needs to contain one or more scripts called ``<tribename>.lua`` which must follow the same conventions as the files in ``data/tribes/initialization/<tribename>/starting_conditions/*.lua`` (see :doc:lua_tribes_tribes_start_conditions).
```

File: data/tribes/initialization/atlanteans/starting_conditions/discovery.lua [hunk 8]
```diff
@@ -1,5 +1,111 @@
+-- RST
+-- .. _lua_tribes_tribes_start_conditions:
+--
+-- Starting Conditions
+-- ===================
+--
+-- The starting conditions a tribe can use are listed in the tribes :ref:`init.lua <lua_tribes_tribes_init>` file.
```
Comment (Noordfrees) [inline data/tribes/initialization/atlanteans/starting_conditions/discovery.lua]:
```suggestion
-- The starting conditions a tribe can use are listed in the tribe’s :ref:`init.lua <lua_tribes_tribes_init>` file.
```

File: data/tribes/initialization/atlanteans/starting_conditions/discovery.lua [hunk 9]
```diff
@@ -1,5 +1,111 @@
...
+--
+-- They are defined in their definition files named as follows and located in
+-- ``data/tribes/initialization/<tribe_name>/starting_conditions/<start_condition_name>.lua``.
```
Comment (Noordfrees) [inline data/tribes/initialization/atlanteans/starting_conditions/discovery.lua]:
```suggestion
-- ``data/tribes/initialization/<tribe_name>/starting_conditions/<starting_condition_name>.lua``.
```

File: data/tribes/initialization/atlanteans/starting_conditions/discovery.lua [hunk 10]
```diff
@@ -1,5 +1,111 @@
...
+--
+-- .. function:: init
+--
+--    This function initializes the starting condition.
+--
+--    **descname**: The localized name of the starting condition.
+--
+--    **tooltip**: A short description of the starting condition. Should contain essential hints if necessary.
+--
+--    **map_tags**: *Optional* A table with map tags that are required for this starting condition.
```
Comment (Noordfrees) [inline data/tribes/initialization/atlanteans/starting_conditions/discovery.lua]:
```suggestion
--    **map_tags**: *Optional* A table with map tags that are required for this starting condition,
```

File: data/tribes/initialization/atlanteans/starting_conditions/discovery.lua [hunk 11]
```diff
@@ -1,5 +1,111 @@
...
+--                  e.g. ``{ "seafaring" }`` if the starting conditions needs a seafaring map.
+--                  If present, the starting condition can only be selected by the user if the map contains *all* listed tags.
+--
+--    **incompatible_wc**: *Optional* A table with strings containing the paths of win conditions that are incompatible with this
+--                         starting condition, e.g. ``{ "scripting/win_conditions/hq_hunter.lua" }`` if the starting condition
+--                         is incompatible with the win condition "HQ Hunter".
+--
+--    **func**: A standardized function to determine whether to share this starting condition between players (Shared Kingdom)
```
Comment (Noordfrees) [inline data/tribes/initialization/atlanteans/starting_conditions/discovery.lua]:
```suggestion
--    **func**: A standardized function to determine whether to share this starting condition between players (Shared Kingdom).
```

File: data/tribes/initialization/atlanteans/starting_conditions/discovery.lua [hunk 12]
```diff
@@ -1,5 +1,111 @@
...
+--              It needs to be declared like this: ``func = function(player, shared_in_start)``.
+--              Later in the code it is necessary to use the result (Variable ``shared_in_start``) to share the starting field
```
Comment (Noordfrees) [inline data/tribes/initialization/atlanteans/starting_conditions/discovery.lua]:
```suggestion
--              Later in the code it is necessary to use the result (variable ``shared_in_start``) to share the starting field
```

File: data/tribes/initialization/atlanteans/starting_conditions/discovery.lua [hunk 13]
```diff
@@ -1,5 +1,111 @@
...
+--              together with the start condition.
+--              Example::
+--
+--                 if shared_in_start then
+--                    sf = shared_in_start
+--                 else
+--                    player:allow_workers("all")
+--                 end
+--
+-- After these declarations the lua commands to define the initial mapobjects for the player follow.
```
Comment (Noordfrees) [inline data/tribes/initialization/atlanteans/starting_conditions/discovery.lua]:
```suggestion
-- After these declarations the Lua function calls to create the initial map objects for the player follow.
```

**Review by Noordfrees — COMMENTED** [review]
A few more nits, then this is good to go.

Code for the HQ hunter bug LGTM, not tested.

File: data/tribes/initialization/atlanteans/starting_conditions/discovery.lua [hunk 14]
```diff
@@ -1,5 +1,111 @@
...
+--                  If present, the starting condition can only be selected by the user if the map contains *all* listed tags.
+--
+--    **incompatible_wc**: *Optional* A table with strings containing the paths of win conditions that are incompatible with this
+--                         starting condition, e.g. ``{ "scripting/win_conditions/hq_hunter.lua" }`` if the starting condition
+--                         is incompatible with the win condition "HQ Hunter".
+--
+--    **func**: A standardized function to determine whether to share this starting condition between players (Shared Kingdom)
+--              It needs to be declared like this: ``func = function(player, shared_in_start)``.
+--              Later in the code it is necessary to use the result (Variable ``shared_in_start``) to share the starting field
+--              together with the start condition.
```
Comment (Noordfrees) [inline data/tribes/initialization/atlanteans/starting_conditions/discovery.lua]:
```suggestion
--              together with the starting condition.
```

**Review by Noordfrees — APPROVED** [review]

**Comment by hessenfarmer** [comment]
@frankystone have you been able to test whether the linked bug gets fixed with this PR?

**Comment by frankystone** [comment]
No, sorry.

**Comment by hessenfarmer** [comment]
Tested it myself. Works as expected. Here is the replay


[replays.zip](https://github.com/widelands/widelands/files/5991409/replays.zip)

**Pull request merged**